ktor-utils/common/src/io/ktor/util/pipeline/Pipeline.kt

```diff
@@ -21,6 +22,11 @@ public open class Pipeline<TSubject : Any, TContext : Any>(
      */
     public val attributes: Attributes = Attributes(concurrent = true)
 
+    /**
+     * Indicated if debug mode is enabled. In debug mode users will get more details in the stacktrace.
+     */
+    public open val developmentMode: Boolean = false
```

Comment:
Isn't it better to mark it as `@Internal`? Otherwise, what is the purpose of this flag for end-users? While it looks ok in config and environment, in a pipeline it doesn't look logical and seems to be some kind internal stuff. 

Also, it will not work if someone creates his own pipelines as he doesn't override it. Hopefully, nobody does it.

And the last question: should we check for `IS_DEVELOPMENT_MODE` instead of just `false`?

Reply:
Yep, let's mark it with internal with error level.

Reply:
If we check `IS_DEVELOPMENT_MODE`, it can use the slow version of the pipeline context accidentally. It can be the cause of errors if someone is implementing the pipeline outside of the Ktor, but I don't insist. What do you think?